cpp/src/binaryop/compiled/util.cpp

```diff
@@ -26,47 +26,60 @@
 namespace cudf::binops::compiled {
 
 namespace {
-/**
- * @brief Functor that returns optional common type of 2 or 3 given types.
- *
- */
+
 struct common_type_functor {
```

Comment:
This is similar to the old code, except we remove a layer of dispatch (2 instead of 3 types). We make up for this in `get_common_type` by doing multiple double-dispatches with this functor in a way that is equivalent to the rules of `std::common_type_t` for multiple input types.